x/dispensation/keeper/distributionRecords.go

```diff
@@ -159,5 +231,68 @@ func (k Keeper) GetRecords(ctx sdk.Context) *types.DistributionRecords {
 		k.cdc.MustUnmarshalBinaryBare(bytesValue, &dr)
 		res.DistributionRecords = append(res.DistributionRecords, &dr)
 	}
+	iterator = k.GetDistributionRecordsIterator(ctx, types.DistributionStatus_DISTRIBUTION_STATUS_FAILED)
+	for ; iterator.Valid(); iterator.Next() {
+		var dr types.DistributionRecord
+		bytesValue := iterator.Value()
+		k.cdc.MustUnmarshalBinaryBare(bytesValue, &dr)
+		res.DistributionRecords = append(res.DistributionRecords, &dr)
+	}
```

Comment:
Do all records not share a common prefix regardless of status, that can be used here? Isn't it possible to have status = unspecified that isn't iterated here?

Reply:
No , records are in different prefixes . This was done to avoid unnessary iteration over Completed and Failed records when running dispesations .
The unspecified status is never set by us